### ISSUE TITLE
[FEATURE] Support Python 3.12 for the C extension

### ISSUE THREAD

**Issue by BwL1289** [Marco-Sulla/python-frozendict#105]
Related to #68.

Please add support for Python 3.12 to the C extension.

OS version: Amazon Linux 2023 (Fedora Based
- Python3 version (python3 -V -V): 3.12.5

Steps to reproduce:
- Attempt to build a wheel

**Comment by Marco-Sulla** [comment]
Yes, in my little free time (I'm a dad now) I'm working on making the C extension available for 3.11. 

This time I want to be close as much as possible to the CPython code, so no more two .c files, one for frozendict code and one for dict code I need to change. It was ideally more clean, but it's more difficult to diff.

Once made 3.11, 3.12 seems to be not so hard. 3.13 will be a nightmare.

**Comment by BwL1289** [comment]
Understood (and congrats on being a dad 👍). Appreciate your help and support.
